### PR TITLE
MINT-1582 Voting redesign

### PR DESCRIPTION
**Pull request by wAISw**
Start implement new design form poll edit form.

### CODE REVIEW

File: packages/core/redux/assetsHolder/selectors/models.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+/**
+ * Copyright 2017–2018, LaborX PTY
+ * Licensed under the AGPL Version 3 license.
+ */
+
+import { DUCK_ASSETS_HOLDER } from '../actions'
+
+/**
```
Comment (ozalexo) [inline packages/core/redux/assetsHolder/selectors/models.js]: Please remove comments like this or replace it with
```
//#region Text description
export yourCode () => {}
//#endregion
```

File: packages/core/redux/session/selectors/models.js [hunk 2]
```diff
@@ -19,3 +19,8 @@ export const getGasSliderCollection = (state) => {
   const { gasPriceMultiplier } = state.get(DUCK_SESSION)
   return gasPriceMultiplier
 }
+
+export const getIsCBE = (state) => {
+  const { isCBE } = state.get(DUCK_SESSION)
+  return isCBE
```
Comment (ozalexo) [inline packages/core/redux/session/selectors/models.js]: Why not `return state.get(DUCK_SESSION)`?

File: packages/core/redux/voting/actions.js [hunk 3]
```diff
@@ -83,20 +90,23 @@ export const createPoll = (poll: PollDetailsModel) => async (dispatch) => {
...
-    // eslint-disable-next-line
-    console.error('create poll error', e.message)
     dispatch(handlePollRemoved(stub.id()))
   }
 }
 
-export const removePoll = (poll: PollDetailsModel) => async (dispatch) => {
+export const removePoll = (pollObject: PTPoll) => async (dispatch, getState) => {
+  let poll = pollObject && pollObject.id
+    ? getState().get(DUCK_VOTING).list().item(pollObject.id)
```
Comment (ozalexo) [inline packages/core/redux/voting/actions.js]: `getState().get(DUCK_VOTING)` must be replaced with appropriate selector from `selectors/models.js`.

File: packages/core/redux/voting/selectors.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    if (objA === objB) {
+      return true
+    }
+
+    if (typeof objA !== 'object' || objA === null ||
+      typeof objB !== 'object' || objB === null) {
+      return false
+    }
+
+    let keysA = Object.keys(objA)
```
Comment (ozalexo) [inline packages/core/redux/voting/selectors.js]: const instead of let

File: packages/core/redux/voting/selectors.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    let keysB = Object.keys(objB)
+
+    if (keysA.length !== keysB.length) {
+      return false
+    }
+
+    // Test for A's keys different from B.
+    let bHasOwnProperty = hasOwnProperty.bind(objB)
```
Comment (ozalexo) [inline packages/core/redux/voting/selectors.js]: const instead of let

File: src/components/dialogs/PublishPollDialog/lang.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+ * Copyright 2017–2018, LaborX PTY
+ * Licensed under the AGPL Version 3 license.
+ */
+
+export const prefix = 'PublishPollDialog'
+
+export default {
+  en: {
+    title: 'Publish this poll?',
+    text: 'You\'re about to publish "%{name}" poll and make it visible to a public. Please confirm blow.',
```
Comment (ozalexo) [inline src/components/dialogs/PublishPollDialog/lang.js]: 'You\'re about to publish "%{name}" poll and make it visible to a public. Please confirm blow.'

Hmmm.. Maybe 'confirm bElow'?

File: src/components/voting/PollDepositWarningWidget/lang.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+ * Licensed under the AGPL Version 3 license.
+ */
+
+export const prefix = 'PollDepositWarningWidget'
+
+export default {
+  en: {
+    warning1: 'Deposit TIME on ',
+    warning2: 'Deposit page',
+    warning3: ', if you want get access this page.',
```
Comment (ozalexo) [inline src/components/voting/PollDepositWarningWidget/lang.js]: if you want get access [to?] this page.

File: src/components/voting/PollEditForm/PollEditForm.jsx [hunk 8]
```diff
@@ -0,0 +1,333 @@
...
+    this.timeout = setTimeout(() => {
+      estimateGasForVoting(
+        action,
+        [action, [optionsSize, 'hashStub', new BigNumber(voteLimitInTIME), newDeadline.getTime()], new BigNumber(0)],
+        (error, { gasFee, gasPrice }) => {
+          if (!error) {
+            this.setState({ gasFee, gasPrice })
+          } else {
+            // eslint-disable-next-line
+            console.log(error)
```
Comment (ozalexo) [inline src/components/voting/PollEditForm/PollEditForm.jsx]: -> log.error

File: src/components/voting/PollEditForm/PollEditForm.scss [hunk 9]
```diff
@@ -2,48 +2,62 @@
...
+    color: $color-white;
+    font-weight: 500;
+    padding: 30px;
   }
 }
 
 .content {
   max-width: 928px;
   background-color: $color-white;
+  font-family: proxima-nova, sans-serif;
```
Comment (ozalexo) [inline src/components/voting/PollEditForm/PollEditForm.scss]: Please remove default font proxima-nova.

File: src/components/voting/PollEditForm/lang.js [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+    pollDescription: 'Description',
+    voteLimit: 'Vote Limit in TIME per option',
+    finishedDate: 'Date',
+    finishedTime: 'Time',
+    addAttachments: 'Attach new',
+    options: 'Options',
+    option: 'Option description',
+    optionIndex: 'Option #%{index}',
+    updatePoll: 'Update Poll',
+    createPoll: 'Create Poll',
```
Comment (ozalexo) [inline src/components/voting/PollEditForm/lang.js]: [Question]

At line 10 we have `title: 'Create New Poll',`
Do we really need 'Create Poll' and 'Create New Poll' at the same time?

File: src/layouts/partials/NewPollContent/NewPollContent.jsx [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+
+import { PollEditForm } from 'components'
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import { initAssetsHolder } from '@chronobank/core/redux/assetsHolder/actions'
+import { listPolls } from '@chronobank/core/redux/voting/actions'
+import './NewPollContent.scss'
+
+function mapStateToProps (state) {
+  return {}
```
Comment (ozalexo) [inline src/layouts/partials/NewPollContent/NewPollContent.jsx]: Do we need empty mapStateToProps?

File: src/layouts/partials/NewPollContent/NewPollContent.jsx [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+function mapDispatchToProps (dispatch) {
+  return {
+    getList: () => dispatch(listPolls()),
```
Comment (ozalexo) [inline src/layouts/partials/NewPollContent/NewPollContent.jsx]: These methods do not use anywhere.
Looks like @connect may be safely removed.

File: src/pages/lib/PollPage.scss [hunk 13]
```diff
@@ -0,0 +1,10 @@
+/**
+ * Copyright 2017–2018, LaborX PTY
+ * Licensed under the AGPL Version 3 license.
+ */
+
+@import "~styles/partials/variables";
+@import "~styles/partials/mixins";
+
+.root {
```
Comment (ozalexo) [inline src/pages/lib/PollPage.scss]: Why this file is so empty?

File: packages/core/redux/assetsHolder/selectors/models.js [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+ * SIMPLE SELECTORS
+ * ==============================================================================
+ */
+
+export const getAssetsHolderAssets = (state) => {
```
[the remark this answers is not in the document]
Comment (wAISw) [reply packages/core/redux/assetsHolder/selectors/models.js]: Removed!

File: packages/core/redux/session/selectors.js [hunk 15]
```diff
@@ -9,3 +9,4 @@ export { getGasSliderCollection } from './selectors/models'
 export { getGasPriceMultiplier }  from './selectors/session'
 export { getAddressesList }from './selectors/session'
 export { getBlockchainAddressesList } from './selectors/session'
+export { isCBE } from './selectors/session'
```
[the remark this answers is not in the document]
Comment (wAISw) [reply packages/core/redux/session/selectors.js]: export * from ..... doesn't work for us.

File: src/components/voting/PollEditForm/lang.js (hunk 10 above)
Comment (wAISw) [reply src/components/voting/PollEditForm/lang.js]: Fixed!

File: packages/core/redux/session/selectors.js (hunk 15 above)
[the remark this answers is not in the document]
Comment (ozalexo) [reply packages/core/redux/session/selectors.js]: Like this:
```export { a, b, c } from 'blaModule'```

**Pull request merged**